superset/datasets/dao.py

```diff
@@ -35,6 +37,26 @@ class DatasetDAO(BaseDAO):  # pylint: disable=too-many-public-methods
     model_cls = SqlaTable
     base_filter = DatasourceFilter
 
+    @classmethod
+    def find_by_ids(cls, model_ids: Union[List[str], List[int]]) -> List[SqlaTable]:
```

Comment:
Is this related?

Reply:
Looks unrelated... maybe this was supposed to be in another branch?

Reply:
I also mentioned it in the PR description, but yes tests were failing on dataset deletion due to sqlalchemy not having access to the database relationship in the session. I found other logs of this happening for this command, and fixed it here for the tests to pass. There was a query in the after update hook that I removed that was re-loading the database into the session as an unexpected side-effect.